### PR TITLE
Harden simulation CLI contracts and error reporting

### PR DESCRIPTION
**Pull request by pdj555**
## Summary
- Make `simulate_cli.run()` headless by default and fail fast when `render=True` is requested without a renderer.
- Reject GBM parameter estimation from fewer than two return observations.
- Surface command-specific guidance from the public CLI on `PriceDataError` / `ValueError`, exit `1` (not silent) when every ticker fails, and log per-ticker errors so operators can recover.
- Refresh `CLAUDE.md` to reflect the current module surface (decision layer, AI summaries, browser UI, output artifacts, CI workflows).
- Add regression tests for the renderer contract, the GBM 2-observation minimum, and the no-results error path.

## Test plan
- [x] `pytest -q` — 128 passed, 1 skipped (Flask install-detection test).
- [x] `flake8 public_cli.py simulate_cli.py simulation.py tests/test_cli.py tests/test_simulation.py` — clean.
- [x] Confirmed the existing `test_public_simulate_reports_missing_offline_data` path still asserts exit `1` with actionable `--data-path` guidance.

### CODE REVIEW

File: simulate_cli.py [hunk 1]
```diff
@@ -297,12 +297,15 @@ def maybe_show_simulation_plots(args: argparse.Namespace, result: dict[str, Any]
 def run(
     args: argparse.Namespace,
     *,
-    render: bool = True,
+    render: bool = False,
```
[the remark this answers is not in the document]
Comment (pdj555) [reply simulate_cli.py:300]: Good catch — fixed in 834a261. Flipped `display_plots` default to `False` in `simulate_cli.run()` so plot display is fully opt-in for programmatic callers using `build_simulation_args()` defaults (where `show=True`). Both CLI entry points already pass `display_plots` explicitly (legacy=True, public=False) so they're unaffected. Added `test_simulate_cli_run_does_not_display_plots_with_default_args` to lock in the contract.
